client/src/main/java/io/pravega/client/stream/impl/ReaderGroupStateManager.java

```diff
@@ -157,18 +159,25 @@ void close() {
     /**
      * Handles a segment being completed by calling the controller to gather all successors to the completed segment.
      */
-    void handleEndOfSegment(Segment segmentCompleted) throws ReinitializationRequiredException {
-        val successors = getAndHandleExceptions(controller.getSuccessors(segmentCompleted), RuntimeException::new);
-        synchronized (this) {
-            latestDelegationToken = successors.getDelegationToken();
+    void handleEndOfSegment(Segment segmentCompleted, boolean fetchSuccesors) throws ReinitializationRequiredException {
```

Comment:
This method should take endSegments into account.

Reply:
As discussed, will not be using endSegments here.